tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -14,33 +14,90 @@ else
 }
 
 
-function Get-AzProfile 
+function Get-ProfileCachePath
 {
-  @{
-    '2015-05' = 
-    @{
-        'AzureRM' = @('1.2.4');
-        'AzureRM.Storage' = @('1.2.4');
-        'Azure.Storage' = @('1.2.4');
-        'AzureRM.Profile' = @('1.2.4');
-     }; 
-     '2016-09' = 
-     @{
-        'AzureRM' = @('2.0.1');
-        'AzureRM.Storage' = @('2.0.1')
-        'Azure.Storage' = @('2.0.1');
-        'AzureRM.Profile' = @('2.0.1');
-     }
-   }
+  $ProfileCache = Join-Path -path $env:LOCALAPPDATA -childpath "Microsoft\AzurePowerShell\ProfileCache"
+  return $ProfileCache
+}
+
+
+# Get-ProfileMap from Azure Endpoint
+function Get-AzureProfileMap
+{
+  $ProfileCache = Get-ProfileCachePath
+  if(-Not (Test-Path $ProfileCache))
+  {
+    New-Item -ItemType Directory -Force -Path $ProfileCache | Out-Null
+  }
+
+  $response = Invoke-RestMethod -ea SilentlyContinue -Uri $PSProfileMapEndpoint -ErrorVariable RestError -OutFile "$ProfileCache\ProfileMap.Json"
+
+  if ($RestError)
+  {
+    $HttpStatusCode = $RestError.ErrorRecord.Exception.Response.StatusCode.value__
+    $HttpStatusDescription = $RestError.ErrorRecord.Exception.Response.StatusDescription
+    
+	Throw "Http Status Code: $($HttpStatusCode) `nHttp Status Description: $($HttpStatusDescription)"
+  }
+
+  $ProfileMap = Get-Content -Raw -Path "$ProfileCache\ProfileMap.json" -ErrorAction SilentlyContinue | ConvertFrom-Json 
+
+  return $ProfileMap
 }
 
 
+function Get-AzProfile
+{
+  [CmdletBinding()]
+  param()
+  DynamicParam
+  {
+    $params = New-Object -Type System.Management.Automation.RuntimeDefinedParameterDictionary
+    Add-ForceParam $params
+    Add-SwitchParam $params "ListAvailable" "ListAvailableParameterSet"
+    return $params
+  }
+
+  PROCESS
+  {
+    $Force = $PSBoundParameters.Force
+    # If force is present, download ProfileMap from Storage blob endpoint
+    if ($Force.IsPresent)
```

Comment:
This is not appropriate use of Force.  Force is used to bypass ShouldContinue prompts, and it means that we need to implement ShouldProcess, so a user would know what would happen if we make any changes.  I think the thing that is intended here is to use a parameter like -Online to grab the map from the latest online version instead of Force